docs/Building.md

```diff
@@ -0,0 +1,127 @@
+`onl-chws`) was actually a home grown Docker-like Linux container based
+system, so this is not a fundamental change in how ONL is built.
+
+
+Build ONL Summary
+------------------------------------------------------------
+    #> git clone git://github.com/opennetworklinux/ONL
+    #> cd ONL
+    #> make docker                                              # enter the docker workspace
+    root@8onl-builder:/path/to/ONL# make onl-x86 onl-powerpc    # build both x86 and PPC images
+
+The resulting ONIE installers are in
+$ONL/builds/installer/$ARCH/all/onl-$VERSION-all.installer,
+and the SWI files (if you want them) are in
+$ONL/builds/swi/$ARCH/all/onl-$VERSION-all.swi.
+
+
+
+
+#Installing Docker Gotchas
+
+Docker installer oneliner (for reference: see docker.com for details)
+
+    # apt-get install -y docker.io
+or
+
+    # wget -qO- https://get.docker.com/ | sh
+
+
+Common docker related issues:
+
+- Check out http://docs.docker.com/installation/ubuntulinux/ for detailed instructions
+- You may have to update your kernel to 3.10+
+- Beware that `apt-get install docker` installs a dock application not docker :-)  You want the docker.io package instead.
+- Some versions of docker are unhappy if you use a local DNS caching resolver:
+	- e.g., you have 127.0.0.1 in your /etc/resolv.conf
+        - if you have this, specify DNS="--dns 8.8.8.8" when you enter the docker environment
+ 	- e.g., `make DNS="--dns 8.8.8.8" docker`
+
+Consider enabling builds for non-privileged users with:
+
+- `sudo usermod -aG docker $USER`
+- If you run as non-root without this, you will get errors like "..: dial unix /var/run/docker.sock: permission denied"	
+- Building as root is fine as well (it immediately jumps into a root build shell), so this optional
+    
+#Additional Build Details
+----------------------------------------------------------
+
+The rest of this guide talks about how to build specific 
+sub-components of the ONL ecosystem and tries to overview
+all of the various elements of the build.
+
+Build all .deb packages for all architectures
+----------------------------------------------------------
+    #> cd $ONL/builds/components
+    #> make
+    #> find $ONL/debian/repo -name \*.deb    # all of the .deb files end up here
+
+A number of things will happen automatically, including:
+
+- git submodule checkouts and updates for kernel, loader, and additional code repositories
+- automatic builds of all debian packages and their dependencies
+- automatic download of binary-only .deb packages from apt.opennetlinux.org
+
+After all components have been built, your can build an ONL
+Software Image from those components.
+
+Adding/Removing packages from a SWI:
+------------------------------------------------------------
+
+The list of packages for a given SWI are in
+    $ONL/builds/swi/$ARCH/all/rootfs/rootfs.$ARCH	# for $ARCH specific packages
+    $ONL/builds/swi/$ARCH/all/rootfs/rootfs		# for $ARCH-independent packages
```

Comment:
The Markdown output collapses these three lines into one. I think you need a blank line after the first line.